src/Strings/ValidParanthesis.java

```diff
@@ -200,6 +200,51 @@ else if(c == '*')
 		}
 		return true;
 	}
+	
+	//saw this solution in leetcode
```

Comment:
I like this solution, no extra space compared to the O(N) space complexity to the above solutions